custom/sensors/SparkEncoder.java

```diff
@@ -0,0 +1,134 @@
+package org.usfirst.frc4904.standard.custom.sensors;
+
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import org.usfirst.frc4904.standard.Util;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class SparkEncoder implements CustomEncoder {
+    protected final CANEncoder encoder;
+    protected PIDSourceType pidSource;
+    protected double distancePerPulse;
+    protected boolean reverseDirection;
+    protected double truePosition;
+    protected double trueVelocity;
+
+    public SparkEncoder(CANSparkMax spark, boolean reverseDirection, double distancePerPulse) {
```

Comment:
Because the encoders return data in rotations, we may want to keep the data unscaled, so there should be a default constructor with distancePerPulse = 1.0